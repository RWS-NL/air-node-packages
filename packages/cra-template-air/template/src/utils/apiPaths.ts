export enum API_METHOD {
  GET = 'get',
  POST = 'post',
  PUT = 'put',
  DELETE = 'delete'
}

export enum ENDPOINTS {
  CONFIG = 'config'
}

export enum API_VERSION {
  v1 = 'v1'
}

export enum ACTION_ENDPOINTS {
  GET_CONFIG = 'getConfig'
}

export const CONFIG_PATH = (version: API_VERSION) => `/api/${version}/${ENDPOINTS.CONFIG}`;
