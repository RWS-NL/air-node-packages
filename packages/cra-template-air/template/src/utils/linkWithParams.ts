import { parse, stringify } from '@favware/querystring';
import windowLocationService from './WindowLocationService';

export const linkWithParams = <T>(
  path: string,
  paramsToAdd: T,
  shouldMerge?: boolean,
  search?: string
) => {
  if (!search) search = windowLocationService.search;

  const currentParams = parse(search);
  let newSearchString = stringify(paramsToAdd);

  if (shouldMerge) {
    newSearchString = stringify({ ...currentParams, ...paramsToAdd });
  }

  if (newSearchString !== '') {
    return `${path}?${newSearchString}`;
  }

  return path;
};

export default linkWithParams;
