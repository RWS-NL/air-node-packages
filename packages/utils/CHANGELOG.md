# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [2.3.1](https://github.com/RWS-NL/air-node-packages/compare/@rws-air/utils@2.3.0...@rws-air/utils@2.3.1) (2020-01-09)

**Note:** Version bump only for package @rws-air/utils

# [2.3.0](https://github.com/RWS-NL/air-node-packages/compare/@rws-air/utils@2.2.1...@rws-air/utils@2.3.0) (2019-12-18)

### Features

- **utils:** add isCIEnv check constant ([8a25b4f](https://github.com/RWS-NL/air-node-packages/commit/8a25b4fe095da054e43512e12f568be990b08769))

## [2.2.1](https://github.com/RWS-NL/air-node-packages/compare/@rws-air/utils@2.2.0...@rws-air/utils@2.2.1) (2019-12-09)

### Bug Fixes

- **utils:** improve mergeObjects typings to support DeepPartial state ([967a8ab](https://github.com/RWS-NL/air-node-packages/commit/967a8ab2280007191d34c6c9ad2ee3eae708e4f0))

# [2.2.0](https://github.com/RWS-NL/air-node-packages/compare/@rws-air/utils@2.1.2...@rws-air/utils@2.2.0) (2019-12-05)

### Features

- **utils:** add mergeobjects function ([cd0bfee](https://github.com/RWS-NL/air-node-packages/commit/cd0bfeeca22011f050b0b3aaab42a65e77041a46))

## [2.1.2](https://github.com/RWS-NL/air-node-packages/compare/@rws-air/utils@2.1.1...@rws-air/utils@2.1.2) (2019-11-28)

### Bug Fixes

- **utils:** fixed stable table sort typings and simplied usage ([296db8d](https://github.com/RWS-NL/air-node-packages/commit/296db8d7dec3ccddaf1e86c9948a9b959e709cc5))

## [2.1.1](https://github.com/RWS-NL/air-node-packages/compare/@rws-air/utils@2.1.0...@rws-air/utils@2.1.1) (2019-11-28)

### Bug Fixes

- **utils:** fixed the export for fuzzysearcher ([5738071](https://github.com/RWS-NL/air-node-packages/commit/57380718d1ec997f3bc55e9e9f1030e2162e6587))

# [2.1.0](https://github.com/RWS-NL/air-node-packages/compare/@rws-air/utils@2.0.0...@rws-air/utils@2.1.0) (2019-11-25)

### Features

- **utils:** add fuzzysearcher ([057b33b](https://github.com/RWS-NL/air-node-packages/commit/057b33b46c339d94d4dee923b508bba0e7f1e030))

# [2.1.0](https://github.com/RWS-NL/air-node-packages/compare/@rws-air/utils@2.0.0...@rws-air/utils@2.1.0) (2019-11-25)

### Features

- **utils:** add fuzzysearcher ([b4b7613](https://github.com/RWS-NL/air-node-packages/commit/b4b761317489e203aa70c8c3355e7329aa1a78eb))

# [2.0.0](https://github.com/RWS-NL/air-node-packages/compare/@rws-air/utils@1.2.0...@rws-air/utils@2.0.0) (2019-11-15)

### Bug Fixes

- **utils:** fixed missing exports ([a78e84a](https://github.com/RWS-NL/air-node-packages/commit/a78e84a59efae795b4618702f143b8580e8d9abf))

### Features

- rework modal ([#143](https://github.com/RWS-NL/air-node-packages/issues/143)) ([fd0da0c](https://github.com/RWS-NL/air-node-packages/commit/fd0da0c38de50e687c36e6b1a547321eb085a2cd))

### BREAKING CHANGES

- Modal has been completely changed and takes different props. This completely breaks any pre-existing instances of modal usage.

Signed-off-by: Kevin Hendriks <kevin.hendriks@cgi.com>

- ci(workflows): add github action to update docs

- build(deps): bump versions

- ci(actions): add label sync

- docs(github): update templates

# [1.2.0](https://github.com/RWS-NL/air-node-packages/compare/@rws-air/utils@1.1.1...@rws-air/utils@1.2.0) (2019-11-12)

### Features

- **utils:** new module: TimeUtils ([8664bbe](https://github.com/RWS-NL/air-node-packages/commit/8664bbea9e04d0f7ece9cd03bcf2795446de9c3f))

## [1.1.1](https://github.com/RWS-NL/air-node-packages/compare/@rws-air/utils@1.1.0...@rws-air/utils@1.1.1) (2019-11-06)

**Note:** Version bump only for package @rws-air/utils

# 1.1.0 (2019-11-01)

### Features

- **utils:** new package: utils ([5660f87](https://github.com/RWS-NL/air-node-packages/commit/5660f8797182c99e6e063f4cbd668bcc1413f85a))
