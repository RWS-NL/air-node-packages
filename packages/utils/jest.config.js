module.exports = {
  projects: [
    {
      displayName: 'ts-jest',
      testMatch: ['<rootDir>/__tests__/**/*.ts'],
      preset: 'ts-jest',
      testEnvironment: 'node',
      testRunner: 'jest-circus/runner'
    },
    {
      displayName: 'react-test',
      testMatch: ['<rootDir>/__tests__/**/*.tsx'],
      snapshotSerializers: ['enzyme-to-json/serializer'],
      setupFiles: ['<rootDir>/jest.setup.ts']
    }
  ]
};
