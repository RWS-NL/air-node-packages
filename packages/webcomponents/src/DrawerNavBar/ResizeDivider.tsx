import Divider, { DividerProps as MDividerProps } from '@material-ui/core/Divider';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import { stringify } from 'flatted';
import React, { memo, useCallback, useEffect, useState } from 'react';
import { TreeDrawerComponentProps } from './DrawerProps';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    resizeDivider: {
      cursor: 'col-resize',
      width: theme.spacing(0.5),
      zIndex: theme.zIndex.drawer + 1,
      backgroundColor: '#999999',
      '&:hover': {
        backgroundColor: '#696969'
      },

      left: ({ treeDrawer, treeDrawerElement }: StyledProps) => {
        if (treeDrawerElement === null)
          treeDrawerElement = document.querySelector('[data-selector="tree-drawer"] > div');

        if (treeDrawerElement !== null && treeDrawerElement.scrollHeight > treeDrawerElement.clientHeight) {
          return `calc(${treeDrawer.width}px - ${theme.spacing(1.5)}px)`;
        }
        return `calc(${treeDrawer.width}px + ${theme.spacing(0.5)}px)`;
      }
    }
  })
);

type StyledProps = Pick<ResizeDividerProps, 'treeDrawer'> & {
  treeDrawerElement: Element | null;
};

export type ResizeDividerProps = Omit<TreeDrawerComponentProps, 'navigationDrawer'> & MDividerProps;

export const ResizableDivider = memo(
  ({ treeDrawer, onMouseDown, onTouchStart, ...props }: ResizeDividerProps) => {
    useWindowResize();
    const treeDrawerElement = useCallback(() => document.querySelector('[data-selector="tree-drawer"] > div'), []);
    const classes = useStyles({
      treeDrawer,
      treeDrawerElement: treeDrawerElement()
    });

    return (
      <Divider
        {...props}
        data-qa='draggable-divider'
        onMouseDown={onMouseDown}
        onTouchStart={onTouchStart}
        classes={{ root: classes.resizeDivider }}
        orientation='vertical'
        draggable
        absolute
        light
      />
    );
  },
  (prevProps, nextProps) => {
    if (prevProps.treeDrawer.open !== nextProps.treeDrawer.open) return false;
    if (prevProps.treeDrawer.width !== nextProps.treeDrawer.width) return false;
    if (stringify(prevProps.treeDrawer) !== stringify(nextProps.treeDrawer)) return false;

    return true;
  }
);

function useWindowResize() {
  const isClient = typeof window === 'object';

  function getSize() {
    return {
      width: isClient ? document.querySelector('[data-selector="tree-drawer"] > div')?.scrollHeight : undefined,
      height: isClient ? document.querySelector('[data-selector="tree-drawer"] > div')?.clientHeight : undefined
    };
  }

  const [windowSize, setWindowSize] = useState(getSize);

  useEffect(() => {
    if (!isClient) {
      return undefined;
    }

    function handleResize() {
      setWindowSize(getSize());
    }

    window.addEventListener('resize', handleResize);
    return () => window.removeEventListener('resize', handleResize);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []); // Empty array ensures that effect is only run on mount and unmount

  return windowSize;
}
