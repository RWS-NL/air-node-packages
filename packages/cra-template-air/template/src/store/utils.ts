import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { ACTION_ENDPOINTS, API_METHOD } from 'utils/apiPaths';
import ApiClient, { FetchResponse, FetchResultTypes } from 'utils/apiClient';
import { call, put } from 'redux-saga/effects';
import { AIRHandledError, HTTP_SUCCESS_CODES, defaultToastrOptions } from 'utils/constants';
import { BAD_REQUEST, FORBIDDEN, NOT_FOUND, REQUEST_TIMEOUT } from 'http-status';
import { actions as toastrActions, toastType } from 'react-redux-toastr';
import i18n from 'config/i18n';

export const composeEnhancers = composeWithDevTools({});

/**
 * Gets some data from the backend
 * @param path The API path to get data from
 * @returns the data from the backend in JSON format
 */
export function* getJson<T>(path: string) {
  const response: FetchResponse<T> = yield call([ApiClient, 'fetch'], API_METHOD.GET, path, FetchResultTypes.JSON);

  if (!response) throw new AIRHandledError(REQUEST_TIMEOUT, 'timed_out_no_response');

  if (!Object.keys(response.data).length && !HTTP_SUCCESS_CODES.includes(response.status)) {
    throw new AIRHandledError(response.status || BAD_REQUEST, 'bad_request');
  }

  return response.data;
}

/**
 * Posts some data to an endpoint on the backend
 * @param path The API path to post data to
 * @param body The object to include as the body of the post
 * @returns The body returned by the backend in JSON format
 */
export function* postJson<T>(path: string, body: any) {
  const response: FetchResponse<T> = yield call(
    [ApiClient, 'fetch'],
    API_METHOD.POST,
    path,
    FetchResultTypes.JSON,
    body
  );

  if (!response) throw new AIRHandledError(REQUEST_TIMEOUT, 'timed_out_no_response');

  if (!Object.keys(response.data).length && !HTTP_SUCCESS_CODES.includes(response.status)) {
    throw new AIRHandledError(response.status || BAD_REQUEST, 'bad_request');
  }

  return response.data;
}

/**
 * Puts some data to an endpoint on the backend
 * @param path The API path to put data to
 * @param body The object to include as the body of the put
 * @returns The body returned by the backend in JSON format
 */
export function* putJson<T>(path: string, body: any) {
  const response: FetchResponse<T> = yield call(
    [ApiClient, 'fetch'],
    API_METHOD.PUT,
    path,
    FetchResultTypes.JSON,
    body
  );

  if (!response) throw new AIRHandledError(REQUEST_TIMEOUT, 'timed_out_no_response');

  if (response.data && !Object.keys(response.data).length && !HTTP_SUCCESS_CODES.includes(response.status)) {
    throw new AIRHandledError(response.status || BAD_REQUEST, 'bad_request');
  }

  return response.data;
}

/**
 * Generates a toastr message
 * @param message The message to show in the toastr
 * @param type Optional type of the toastr
 */
export const displayToastr = (message: string, type: toastType = 'success') => {
  return toastrActions.add({
    ...defaultToastrOptions,
    message,
    type
  });
};

/**
 * Handles HTTP status codes by generating toastrs
 * @param code The HTTP status code that was returned
 * @param action The endpoint where the status code occured on
 */
export function* errorHandler(code: number, action: ACTION_ENDPOINTS) {
  switch (code) {
    case BAD_REQUEST:
      return yield put(displayToastr(i18n.t(`toasts.badRequest.${action}`), 'error'));
    case FORBIDDEN:
      return yield put(displayToastr(i18n.t(`toasts.forbidden.${action}`), 'error'));
    case NOT_FOUND:
      return yield put(displayToastr(i18n.t(`toasts.notFound.${action}`), 'error'));
    case REQUEST_TIMEOUT:
      return yield put(displayToastr(i18n.t(`toasts.requestTimeout.${action}`), 'error'));
    default:
      return yield put(displayToastr(i18n.t('toasts.unhandledError'), 'error'));
  }
}
