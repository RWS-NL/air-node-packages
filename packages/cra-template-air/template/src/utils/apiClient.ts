import { BAD_REQUEST, FORBIDDEN, INTERNAL_SERVER_ERROR, NOT_FOUND, UNAUTHORIZED } from 'http-status';
import { API_METHOD } from './apiPaths';
import { AIRHandledError } from './constants';

export enum FetchResultTypes {
  JSON,
  Blob,
  Text,
  Result
}

export interface FetchResponse<T> {
  data: T;
  status: number;
  statusText: string;
  headers: Headers;
}

export default class ApiClient {
  public static accessToken: string | undefined;
  private static apiUrl: string | undefined;

  public get accessToken(): string | undefined {
    return ApiClient.accessToken;
  }

  public set accessToken(value: string | undefined) {
    ApiClient.accessToken = value;
  }

  public get apiUrl(): string | undefined {
    return ApiClient.apiUrl;
  }

  public set apiUrl(value: string | undefined) {
    ApiClient.apiUrl = value;
  }

  public constructor(apiUrl?: string) {
    this.apiUrl = apiUrl || '';
  }

  public static async fetch(method: API_METHOD, url: string, type: FetchResultTypes.Blob, body?: any): Promise<FetchResponse<Blob>>;
  public static async fetch(method: API_METHOD, url: string, type: FetchResultTypes.Text, body?: any): Promise<FetchResponse<string>>;
  public static async fetch(method: API_METHOD, url: string, type: FetchResultTypes.Result, body?: any): Promise<Response>;
  public static async fetch<T extends unknown>(method: API_METHOD, url: string, type: FetchResultTypes.JSON, body?: any): Promise<FetchResponse<T>>;
  public static async fetch<T extends unknown>(method: API_METHOD, url: string, type: FetchResultTypes, body?: any): Promise<Response | FetchResponse<Blob> | FetchResponse<string> | FetchResponse<T>>;
  public static async fetch<T extends unknown>(method: API_METHOD, url: string, type: FetchResultTypes, body?: any) {
    try {
      const response: Response = await fetch(`${ApiClient.apiUrl || ''}${url}`, {
        method,
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Bearer ${ApiClient.accessToken}`
        },
        body: JSON.stringify(body)
      });

      if (!response.ok) throw new AIRHandledError(response.status, `${response.status} error`);

      switch (type) {
        case FetchResultTypes.Result:
          return response;
        case FetchResultTypes.Blob:
          return {
            data: await response.blob(),
            status: response.status,
            statusText: response.statusText,
            headers: response.headers
          } as FetchResponse<Blob>;
        case FetchResultTypes.JSON:
          return {
            data: await ApiClient.parseBody<T>(response),
            status: response.status,
            statusText: response.statusText,
            headers: response.headers
          } as FetchResponse<T>;
        case FetchResultTypes.Text:
          return {
            data: await response.text(),
            status: response.status,
            statusText: response.statusText,
            headers: response.headers
          } as FetchResponse<string>;
        default:
          throw new TypeError(`Unknown fetch type ${type}`);
      }
    } catch (err) {
      if (err.toString().includes(BAD_REQUEST.toString())) throw new AIRHandledError(BAD_REQUEST, '400 error');
      if (err.toString().includes(UNAUTHORIZED.toString())) throw new AIRHandledError(UNAUTHORIZED, '401 error');
      if (err.toString().includes(FORBIDDEN.toString())) throw new AIRHandledError(FORBIDDEN, '403 error');
      if (err.toString().includes(NOT_FOUND.toString())) throw new AIRHandledError(NOT_FOUND, '404 error');

      throw new AIRHandledError(INTERNAL_SERVER_ERROR, 'Internal Fetch Error');
    }
  }

  private static async parseBody<T>(response: Response): Promise<T | undefined> {
    const contentType = response.headers.get('content-type');

    if (contentType && contentType.includes('application/json')) {
      return (await response.json()) as Promise<T>;
    }

    return undefined;
  }

  public async fetch(method: API_METHOD, url: string, type: FetchResultTypes.Blob, body?: any): Promise<FetchResponse<Blob>>;
  public async fetch(method: API_METHOD, url: string, type: FetchResultTypes.Text, body?: any): Promise<FetchResponse<string>>;
  public async fetch(method: API_METHOD, url: string, type: FetchResultTypes.Result, body?: any): Promise<Response>;
  public async fetch<T extends unknown>(method: API_METHOD, url: string, type: FetchResultTypes.JSON, body?: any): Promise<FetchResponse<T>>;
  public async fetch<T extends unknown>(method: API_METHOD, url: string, type: FetchResultTypes, body?: any): Promise<Response | FetchResponse<Blob> | FetchResponse<string> | FetchResponse<T>>;
  public async fetch<T extends unknown>(method: API_METHOD, url: string, type: FetchResultTypes, body?: any) {
    return ApiClient.fetch<T>(method, url, type, body);
  }
}
