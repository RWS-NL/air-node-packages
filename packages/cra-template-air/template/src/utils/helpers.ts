import { capitilizeFirstLetter, snakeToCamelCase } from '@rws-air/utils';
import { Oauth2Permissions, UserPrivileges } from 'store/oauth2/oauth2Types';

export const noOp = () => undefined;

export const transformPrivileges = (privileges: UserPrivileges) =>
  privileges
    .map((entry) => `can${capitilizeFirstLetter(snakeToCamelCase(entry.toLowerCase()))}`)
    .reduce<Oauth2Permissions>((prevVal, curVal) => ({ [curVal]: true, ...prevVal }), {});
