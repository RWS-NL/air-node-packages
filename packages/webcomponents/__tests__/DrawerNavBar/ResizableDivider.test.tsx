import { ResizableDivider } from '@DrawerNavBar/ResizeDivider';
import { treeDrawerProps } from '@Resource/commonDrawerNavBarProps';
import { findShallowByDataQa, toBeCalled } from '@rws-air/utils';
import { shallow } from 'enzyme';
import React from 'react';

const mockOnMouseDown = jest.fn();
const mockOnTouchStart = jest.fn();

describe('ResizableDivider', () => {
  describe('Component Testing', () => {
    const resizableDivider = shallow(
      <ResizableDivider treeDrawer={treeDrawerProps} onMouseDown={mockOnMouseDown} onTouchStart={mockOnTouchStart} />
    );

    test('GIVEN mouseDown event on draggable divider THEN triggers function', () => {
      findShallowByDataQa(resizableDivider, 'draggable-divider').simulate('mousedown');

      toBeCalled(mockOnMouseDown, 1);
    });

    test('GIVEN touchStart event on draggable divider THEN triggers function', () => {
      findShallowByDataQa(resizableDivider, 'draggable-divider').simulate('touchstart');

      toBeCalled(mockOnTouchStart, 1);
    });
  });

  describe('Snapshot Testing', () => {
    test('Required Props', () => {
      const resizableDivider = shallow(
        <ResizableDivider treeDrawer={treeDrawerProps} onMouseDown={mockOnMouseDown} onTouchStart={mockOnTouchStart} />
      );
      expect(resizableDivider).toMatchSnapshot();
    });

    test('Optional Props', () => {
      const resizableDivider = shallow(
        <ResizableDivider
          treeDrawer={treeDrawerProps}
          onMouseDown={mockOnMouseDown}
          onTouchStart={mockOnTouchStart}
          data-qa='testing'
          id='draggable-divider'
        />
      );
      expect(resizableDivider).toMatchSnapshot();
    });
  });
});
