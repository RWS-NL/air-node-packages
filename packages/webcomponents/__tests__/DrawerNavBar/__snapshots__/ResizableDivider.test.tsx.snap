// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResizableDivider Snapshot Testing Optional Props 1`] = `
<WithStyles(ForwardRef(Divider))
  absolute={true}
  classes={
    Object {
      "root": "makeStyles-resizeDivider-1 makeStyles-resizeDivider-4",
    }
  }
  data-qa="draggable-divider"
  draggable={true}
  id="draggable-divider"
  light={true}
  onMouseDown={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onTouchStart={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  orientation="vertical"
/>
`;

exports[`ResizableDivider Snapshot Testing Required Props 1`] = `
<WithStyles(ForwardRef(Divider))
  absolute={true}
  classes={
    Object {
      "root": "makeStyles-resizeDivider-1 makeStyles-resizeDivider-3",
    }
  }
  data-qa="draggable-divider"
  draggable={true}
  light={true}
  onMouseDown={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onTouchStart={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  orientation="vertical"
/>
`;
